arangod/Aql/SingleRemoteModificationExecutor.cpp

```diff
@@ -109,24 +108,19 @@ bool SingleRemoteModificationExecutor<Modifier>::doSingleRemoteModificationOpera
 
   OperationOptions& options = _info._options;
 
-  if (_info._key.empty() && !_info._input1RegisterId.has_value()) {
+  if (_info._key.empty() && _info._input1RegisterId == ExecutionNode::MaxRegisterId) {
     THROW_ARANGO_EXCEPTION_MESSAGE(TRI_ERROR_ARANGO_DOCUMENT_NOT_FOUND,
                                    "missing document reference");
   }
 
   VPackBuilder inBuilder;
   VPackSlice inSlice = VPackSlice::emptyObjectSlice();
-  if (_info._input1RegisterId.has_value()) {  // IF NOT REMOVE OR SELECT
-    // if (_buffer.size() < 1) {
-    //  THROW_ARANGO_EXCEPTION_MESSAGE(TRI_ERROR_ARANGO_DOCUMENT_NOT_FOUND,
-    //                                 "missing document reference in Register");
-    //}
-    AqlValue const& inDocument = input.getValue(_info._input1RegisterId.value());
+  if (_info._input1RegisterId == ExecutionNode::MaxRegisterId) {  // IF NOT REMOVE OR SELECT
```

Comment:
doesn't this `if` revert the old behavior?
```suggestion
  if (_info._input1RegisterId != ExecutionNode::MaxRegisterId) {  // IF NOT REMOVE OR SELECT
```